File: IntersectionofTwoArrays_leetcode#349.js
var intersection = function (nums1, nums2) {
    let ans = [];
    let shorter;
    let longer;
    if (nums1.length >= nums2.length) {
        shorter = nums2;
        longer = nums1;
    } else {
        shorter = nums1;
        longer = nums2;
    }

    for (let i = 0; i < shorter.length; i++) {
        let num = shorter[i];
        if (longer.includes(num) && !ans.includes(num)) {
            ans.push(num);
        }
    }
    return ans;
};

File: IntersectionofTwoArrays_leetcode#349.rb
def intersection(nums1, nums2)
    ans = []
    if nums1.length >= nums2.length
        longer = nums1
        shorter = nums2
    else
        shorter = nums1
        longer = nums2
    end
    
    shorter.each do |num|
        ans.push(num) if longer.include?(num) && !ans.include?(num)
    end
    ans
end